### PR TITLE
Update tsh bench ssh

### PR DESCRIPTION
**Pull request by rosstimothy**
- Adds support for connecting to a random host
- Removes ability to test interactive sessions since they did not actually open an interactive session.

### CODE REVIEW

File: lib/benchmark/ssh.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		return nil, trace.BadParameter("no target hosts available")
+	}
+
+	return func(ctx context.Context) error {
+		nc, err := tc.ConnectToNode(
+			ctx,
+			clt,
+			client.NodeDetails{Addr: chooseRandomHost(resources), Namespace: tc.Namespace, Cluster: tc.SiteName},
+			tc.Config.HostLogin,
+		)
```
Comment (fspmarshall) [inline lib/benchmark/ssh.go:77]: nit: it feels wrong to skip a non-trivial chunk of the `tc.SSH` code path... the code we're skipping is complex enough that I wonder if we might miss future regressions in there.  Admittedly, we're still skipping plenty of `tsh` code, so maybe the point is moot... but I think we could implement this through `tc.SSH` if we added functional options.  Ex:

```golang
tc.SSH(htx, s.Command, false, client.WithNodeAddr(chooseRandomHost(resources)))
```

Minor nit... even if we did the above, it doesn't stop folks from introducing regressions in `tsh ssh` _prior_ to the call to `tc.SSH`.

**Review by fspmarshall — APPROVED** [review]
nit: IIUC you'll still need to provide a dummy hostname in order for `tc` to be initialized correctly (e.g. `tsh bench ssh --random root@this-value-is-ignored...`).  This feels a bit janky, but switching to explicit `--user` is also weird, because of how tightly coupled arg parsing is with client configuration...

Compromise: let's assert that `tc.HostLogin` ends up being some expected sentinel value that is unlikely to be a real hostname (e.g. `all`)...

```
tsh bench ssh --random <user>@all <command>
```

That way we can keep relying on the standard client arg parsing setup, but without having arbitrary inputs that just get ignored (which IMO ends up being confusing/misleading).

File: lib/benchmark/ssh.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+		return tc.SSH(ctx, s.Command, false)
+	}, nil
+}
+
+// random creates a [WorkloadFunc] that executes the provided command on
+// a random host that the user has access to. If hosts disappear or go
+// offline during the benchmark they will not be removed from the list and
+// can have a negative impact on the results.
+func (s SSHBenchmark) random(ctx context.Context, tc *client.TeleportClient) (WorkloadFunc, error) {
+	clt, err := tc.ConnectToCluster(ctx)
```
Comment (codingllama) [inline lib/benchmark/ssh.go:59]: Suggestion: be more explicit about what client we are getting back?

```suggestion
	clusterClient, err := tc.ConnectToCluster(ctx)
```

File: lib/benchmark/ssh.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	}
+	defer clt.Close()
+
+	filter := tc.DefaultResourceFilter()
+	filter.ResourceType = types.KindNode
+	resources, err := apiclient.GetResourcesWithFilters(ctx, clt.AuthClient, *filter)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
```
Comment (codingllama) [inline lib/benchmark/ssh.go:71]: nit:

```suggestion
```

**Review by codingllama — APPROVED** [review]

**Pull request merged**